pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -135,6 +136,21 @@ type PrefilterPlugin interface {
 	Prefilter(pc *PluginContext, p *v1.Pod) *Status
 }
 
+// FilterPlugin is an interface for Filter plugins. These plugins are called at the
+// filter extension point for filtering out hosts that cannot run a pod.
+// This concept used to be called 'predicate' in the original scheduler.
+// These plugins should return only Success or Error in Status.code. However,
+// the scheduler accepts other valid codes as well. Anything other than Success
+// will lead to exclude hosts from running the pod.
+type FilterPlugin interface {
+	Plugin
+	// Filter is called by the scheduling framework.
+	// All FilterPlugins should return sucecss. Otherwise, the node is filtered out.
+	// If Filter returns error, please refer scheduler/algorithm/predicates/error.go
+	// to set error message.
+	Filter(pc *PluginContext, pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *Status
```

Comment:
similar to other plugins, we should only pass the node name not the nodeinfo. A filter plugin can lookup NodeInfo from NodeInfoSnapshot which exist in the FrameworkHandle.

Reply:
I will revise it to align with other plugins.

Reply:
Take a look at the permit plugin integration tests as an example how a plugin can access the framework handler and the node info snapshot.

Reply:
Got it

The individual plugin should init FrameworkHandle during register itself.

// NewPermitPlugin is the factory for permit plugin.
func NewPermitPlugin(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
	perPlugin.fh = fh
	return perPlugin, nil
}